src/caldav-calendar.c

```diff
@@ -396,7 +396,8 @@ static Event* create_event_from_parsed_xml(CaldavCalendar* cal, CaldavEntry* cde
 	// event updated
 	icalcomponent* comp = icalparser_parse_string(cde->caldata);
 	icalcomponent* vev = icalcomponent_get_first_component(comp, ICAL_VEVENT_COMPONENT);
-	g_assert_nonnull(vev);
+	if (!vev)
+		return NULL;
```

Comment:
Just realised this will leak `comp` (`vev` not leaked as `icalcomponent_get_first_component` doesn't pass ownership). `icalcomponent_free` should be called before returning

Reply:
Okay, thanks for catching this! Easy fix, pushed the change on top of rebasing on latest master.